File: 0941-53-2657-find-the-prefix-common-array-of-two-arrays/0941-53-2657-find-the-prefix-common-array-of-two-arrays.js
var findThePrefixCommonArray = function(A, B) {
    const vis = Array(51).fill(0);
    const vis2 = Array(51).fill(0);
    const result = [];

    for (let i = 0; i < A.length; i++) {
        vis[A[i]] = 1;
        vis2[B[i]] = 1;
        let count = 0;
        for (let j = 0; j <= 50; j++) {
            if (vis[j] && vis2[j]) {
                count++;
            }
        }
        result.push(count);
    }

    return result;
};

File: 0941-53-2657-find-the-prefix-common-array-of-two-arrays/0941-53-2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: list[int], B: list[int]) -> list[int]:
        vis = [0] * 51
        vis2 = [0] * 51
        result = []

        for i in range(len(A)):
            vis[A[i]] = 1
            vis2[B[i]] = 1
            count = sum(vis[j] & vis2[j] for j in range(51))
            result.append(count)

        return result
